sensor/models.py

```diff
@@ -29,6 +29,42 @@ class Location(Model):
     def __unicode__(self):
         return self.name
 
+    @staticmethod
+    def _assert_field_for_create_(dict_key, data, err=None):
+        if dict_key not in data:
+            raise KeyError('Needs %s to create location!' % (err if err else dict_key))
+
+    @classmethod
+    def create(cls, data):
+        _assert_field_for_create('id', data, err='device id')
+        _assert_field_for_create('key', data, err='api key')
+        _assert_field_for_create('latitude', data)
+        _assert_field_for_create('longitude', data)
+        _assert_field_for_create('altitude', data)
+        _assert_field_for_create('name', data)
+
+        dev_id = data["id"]
+        try:
+            device = Device.objects.get( pk = dev_id )
+        except Device.DoesNotExist:
+            raise ValueError('No such device "%s".' % dev_id)
+
+        dev_key = data["key"]
+        if not device.valid_post_key(key):
+            raise ValueError('Invalid post key "%s" for device "%s".' % (dev_key, dev_id))
+
+        lat, lon, alt, name = (data["latitude"], data["longitude"],
+                               data["altitude"], data["name"])
+
+        try:
+            lat = '%.8f' % float(lat)
+            lon = '%.8f' % float(lon)
+            alt = '%d' % int(float(alt))
+        except ValueError as err:
+            raise ValueError('Latitude, longitude, and altitude must be numbers. %s' % err)
```

Comment:
This should be handled by the Django layer, no need to do manual numerical conversions.